app/controllers/search_controller.rb

```diff
@@ -0,0 +1,38 @@
+class SearchController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @locations = Location.all
+    @categories = Category.all
+    if params[:search] == nil && params[:location_id] == nil && params[:category] == nil
+      @trucks = Truck.all
+    elsif params[:search] != "" && params[:location_id] != "" && params[:category] != nil
+      like_keyword = "%#{params[:search]}%"
+      trucks1 = Truck.where("lower(name) LIKE ?", like_keyword.downcase).where(location_id: params[:location_id])
+      @trucks = trucks1.joins(categorizations: :category).where(categories: {cuisine: params[:category]})
+    elsif params[:search] != "" && params[:location_id] != ""
+      like_keyword = "%#{params[:search]}%"
+      @trucks = Truck.where("lower(name) LIKE ?", like_keyword.downcase).where(location_id: params[:location_id])
+    elsif params[:location_id] != "" && params[:category] != nil
+      trucks1 = Truck.where(location_id: params[:location_id])
+      @trucks = trucks1.joins(categorizations: :category).where(categories: {cuisine: params[:category]})
+    elsif params[:search] != "" && params[:category] != nil
+      like_keyword = "%#{params[:search]}%"
+      trucks1 = Truck.where("lower(name) LIKE ?", like_keyword.downcase)
+      @trucks = trucks1.joins(categorizations: :category).where(categories: {cuisine: params[:category]})
+    elsif params[:search] != ""
+      like_keyword = "%#{params[:search]}%"
+      @trucks = Truck.where("lower(name) LIKE ?", like_keyword.downcase)
+    elsif params[:location_id] != ""
+      @trucks = Truck.where(location_id: params[:location_id])
+    elsif params[:category]
+      @trucks = Truck.joins(categorizations: :category).where(categories: {cuisine: params[:category]})
+    else
+      @trucks = Truck.all
+    end
+  end
```

Comment:
Two things
- If you really want a search this robust, you might consider using a gem like https://github.com/Casecommons/pg_search
- Regardless, this code is too busy, and should be abstracted away. I would start moving some of this logic to instance methods on the truck object, or at least private methods in this controller

Reply:
Additionally, its up to you, but this method technically does not need its own controller. The resource the it is concerned with is still a Truck, so you could instead define a custom action in the TrucksController called `search` that handles this instead. It would be a more standard process